sound/soc/intel/boards/sof_sdw.c

```diff
@@ -1679,19 +1708,22 @@ static void mc_dailink_exit_loop(struct snd_soc_card *card)
 {
 	struct snd_soc_dai_link *link;
 	int ret;
-	int i, j;
+	int i, j, k;
 
 	for (i = 0; i < ARRAY_SIZE(codec_info_list); i++) {
-		if (!codec_info_list[i].dais[0].exit)
-			continue;
 		/*
 		 * We don't need to call .exit function if there is no matched
 		 * dai link found.
 		 */
 		for_each_card_prelinks(card, j, link) {
-			if (!strcmp(link->codecs[0].dai_name,
-				    codec_info_list[i].dais[0].dai_name)) {
-				ret = codec_info_list[i].dais[0].exit(card, link);
+			for (k = 0; k < codec_info_list[i].dai_num; k++) {
+				if (!codec_info_list[i].dais[k].exit)
+					continue;
+				if (strcmp(link->codecs[0].dai_name,
```

Comment:
not following the [0] here.

Reply:
that's not necessarily true. There could be a scenario where we have different amplifiers connected on the same dai, and one of the amplifiers deals with tweeter and the other one with woofer.

That's a big assumption here!

Reply:
> that's not necessarily true. There could be a scenario where we have different amplifiers connected on the same dai, and one of the amplifiers deals with tweeter and the other one with woofer.
> 
> That's a big assumption here!

Yes, but that's what we have today.

Reply:
> we need to plan for this to be changed. I am 100% sure someone will come up with designs with two different amps just to shave a bit of cost.

Agree, I modified the mc_dailink_exit_loop() function to check dais one by one. Verified with rt1316 + rt1318.